### PR TITLE
[10.0][ADD] web_cache_name_get

### PR DESCRIPTION
**Pull request by hparfr**
This modules cache some requests to name_get() done from list views.

### CODE REVIEW

**Comment by hparfr** [comment]
In order to test this module : 
1) install sales
2) create an SO with a lot of lines and taxes.
3) open the web inspector, network pane and reload the page -> count the name_get()
4) install web_cache_name_get, go back to the same sale order
5) open the web inspector, network pane and reload the page -> count the name_get()

In attachment, an example with 10 sale order lines (each line vat tax at 15%).
web_cache_name_get

**Comment by hparfr** [comment]
@legalsylvain @Yajo

**Comment by yajo** [comment]
You should propose this upstream instead. See https://github.com/odoo/odoo/pull/15967, which fixed a similar bug.

**Review by bealdav — APPROVED** [review]
Ok for me, thanks
nitpicking : application: False is useless

File: web_cache_name_get/README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+================
+Cache name_get()
```
Comment (nikul-serpentcs) [inline web_cache_name_get/README.rst]: @hparfr Name should be `Web Cache Name Get`

File: web_cache_name_get/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Raphaël Reverdy - Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Cache name_get",
+    "version": "10.0.1.0.0",
+    "author": "Akretion, "
```
Comment (nikul-serpentcs) [inline web_cache_name_get/__manifest__.py:9]: @hparfr Added website key as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: web_cache_name_get/views/view.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (nikul-serpentcs) [inline web_cache_name_get/views/view.xml]: Remove `data`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: web_cache_name_get/README.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   :alt: License: LGPL-3
+
+==================
+Web Cache Name Get
+==================
+
+This module improves the loading time of some views: mainly sale or purchase orders 
+with many order lines.
+
+It reduce the numer of useless requests done by the user's browser.
```
Comment (sudhir-erpharbor) [inline web_cache_name_get/README.rst]: reduces
correct the spelling "number"

File: web_cache_name_get/README.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+
+This modules cache some requests to name_get() done from list views.
+
+For instance, if you have 40 sale order lines on a sale. All theses lines have the same tax.
```
Comment (sudhir-erpharbor) [inline web_cache_name_get/README.rst]: All theses lines = All these lines

**Pull request closed without merging**